inference-engine/ie_bridges/python/sample/speech_sample/speech_sample.py

```diff
@@ -0,0 +1,390 @@
+    model.add_argument('-rg', '--import_gna_model', type=str,
+                       help='Read GNA model from file using path/filename provided (required if -m is missing).')
+    args.add_argument('-i', '--input', required=True, type=str, help='Required. Path to an utterance file.')
+    args.add_argument('-o', '--output', type=str, help='Optional. Output file name to save inference results.')
+    args.add_argument('-r', '--reference', type=str,
+                      help='Optional. Read reference score file and compare scores.')
+    args.add_argument('-d', '--device', default='CPU', type=str,
+                      help='Optional. Specify a target device to infer on. '
+                      'CPU, GPU, MYRIAD, GNA_AUTO, GNA_HW, GNA_SW_FP32, GNA_SW_EXACT and HETERO with combination of GNA'
+                      ' as the primary device and CPU as a secondary (e.g. HETERO:GNA,CPU) are supported. '
+                      'The sample will look for a suitable plugin for device specified. Default value is CPU.')
+    args.add_argument('-bs', '--batch_size', default=1, type=int, help='Optional. Batch size 1-8 (default 1).')
+    args.add_argument('-qb', '--quantization_bits', default=16, type=int,
+                      help='Optional. Weight bits for quantization: 8 or 16 (default 16).')
+    args.add_argument('-wg', '--export_gna_model', type=str,
+                      help='Optional. Write GNA model to file using path/filename provided.')
+    # TODO: Find a model that applicable for -we option
+    args.add_argument('-we', '--export_embedded_gna_model', type=str, help=argparse.SUPPRESS)
+    # TODO: Find a model that applicable for -we_gen option
+    args.add_argument('-we_gen', '--embedded_gna_configuration', default='GNA1', type=str, help=argparse.SUPPRESS)
+    args.add_argument('-iname', '--input_layers', type=str,
+                      help='Optional. Layer names for input blobs. The names are separated with ",". '
+                      'Allows to change the order of input layers for -i flag. Example: Input1,Input2')
+    args.add_argument('-oname', '--output_layers', type=str,
+                      help='Optional. Layer names for output blobs. The names are separated with ",". '
+                      'Allows to change the order of output layers for -o flag. Example: Output1:port,Output2:port.')
+
+    return parser.parse_args()
+
+
+def get_scale_factor(matrix: np.ndarray) -> float:
+    """Get scale factor for quantization using utterance matrix"""
+    # Max to find scale factor
+    target_max = 16384
+    max_val = np.max(matrix)
+    if max_val == 0:
+        return 1.0
+    else:
+        return target_max / max_val
+
+
+def read_ark_file(file_name: str) -> dict:
+    """Read utterance matrices from a .ark file"""
+    def read_key(input_file: IO[Any]) -> str:
+        """Read a identifier of utterance matrix"""
+        key = ''
+        while True:
```

Comment:
Please rewrite it  without `while True`

Reply:
I used `while True` because can't know a number of utterances before reading a input file. Also before each matrix there is a header containing information about type (float or double) of its elements.

Reply:
@generalova-kate why we can use  `char` check in condition?

Reply:
@nkogteva A space char is used between a key and a header to separate them. An empty char indicates a end of a file.

Reply:
@dpigasin I see. I'm just trying to say that `while True` is bad code practice. It is better when you use while <condition>. Python has not while do construction, but it is not a problem to initialize a variable by None, for example, and check it in the cycle.

Reply:
@nkogteva ok, I got it

Reply:
we will add it as a separate MR. It is so big